pkg/helm/actions/upgrade_release.go

```diff
@@ -199,11 +195,7 @@ func UpgradeReleaseAsync(
 				return nil, fmt.Errorf("error setting up authentication: %v", err)
 			}
 		}
-		if len(tlsFiles) == 0 {
-			chartLocation = chartUrl
-		} else {
-			chartLocation = chartInfo.Name
-		}
+		chartLocation = chartUrl
 		client.ChartPathOptions.Version = chartInfo.Version
 		cp, err = client.ChartPathOptions.LocateChart(chartLocation, settings)
```

Comment:
Unless we are trying to minimize the changes (and, possibly, even if we're not!...), I recommend removing the `chartLocation` variable:  it no longer has a life which is independent of `chartUrl`, so we should just replace it with the latter.